src/ImageSharp/Processing/Processors/Quantization/EuclideanPixelMap{TPixel}.cs

```diff
@@ -91,16 +105,43 @@ public void Clear(ReadOnlyMemory<TPixel> palette)
         this.cache.Clear();
     }
 
+    /// <summary>
+    /// Allows setting the transparent index after construction.
+    /// </summary>
+    /// <param name="index">An explicit index at which to match transparent pixels.</param>
+    public void SetTransparentIndex(int index) => this.transparentIndex = index;
+
     [MethodImpl(InliningOptions.ShortMethod)]
     private int GetClosestColorSlow(Rgba32 rgba, ref TPixel paletteRef, out TPixel match)
     {
         // Loop through the palette and find the nearest match.
         int index = 0;
         float leastDistance = float.MaxValue;
+
+        if (this.transparentIndex >= 0 && rgba == default)
+        {
+            // We have explicit instructions. No need to search.
+            index = this.transparentIndex;
+            this.cache.Add(rgba, (byte)index);
+
+            if (index >= 0 && index < this.Palette.Length)
```

Comment:
Given that in this branch `index == this.transparentIndex >= 0`, the `index >= 0` condition seems to be unnecessary.

Regarding `index < this.Palette.Length`, shouldn't we make this an invariant rule instead, and `Guard` it in the constructor and `SetTransparentIndex`?

Reply:
I'm not sure I follow. If you have time, could you push to the branch?

Reply:
Will do it later this week.

Reply:
Thanks!

Reply:
b48dbc5087de9bd98dc8ed0f478550e22d4e1ff9

Reply:
Ah yeah, that's much better! Not sure what I was thinking when I wrote mine! 🤣